apps/addons/buttons.py

```diff
@@ -233,11 +234,13 @@ def __init__(self, text, url, os=None, file=None):
 
 # Cache it for a year.
 @cache_page(60 * 60 * 24 * 365)
+@non_atomic_requests
```

Comment:
Which is why I thought it would make more sense at the top, since I consider the view to be the view + all its decorators, but I guess you could argue it the other way around too...

I just know from experience with zamboni that's it tricky to get right, so we should be careful and maybe test that (see below)